app/models/site.rb

```diff
@@ -53,6 +53,30 @@ def to_s
     "#{location_name} (code: #{code})"
   end
 
+  def copy_to_provider(new_provider)
+    new_site = new_provider.sites.find_by(code: self.code)
+    unless new_site
+      new_site = self.dup
+      new_site.provider_id = new_provider.id
+      new_site.save(validate: false)
```

Comment:
is this bit still needed, after #605

Reply:
Ya, it's still needed. `postcode` and `address1` are still blank on some records. 😿